rfcs/20240311-verbose-logging-support/README.md

```diff
@@ -0,0 +1,87 @@
+Specifying `DNNL_LOGFILE` automatically enables logging of the verbose output to the user-specified file while in the default case, the data is directly printed to `stdout`.
+In this scenario, the data recorded in the logfile mirrors the printed verbose mode information, hence, the logged data can be managed using oneDNN [runtime controls](https://oneapi-src.github.io/oneDNN/dev_guide_verbose.html?highlight=onednn_verbose#run-time-controls) for the verbose mode.
+- By default, data will be recorded using a rotating lazy logger with a file size specified by `ONEDNN_LOGFILE_SIZE(=1024*1024)` and the number of rotatiing files specified by DNNL_LOGFILE_AMT(=5).
+
+### 3. Alignment and Specification of Logging Levels
+`spdlog` defines the following levels for data logging in its implementation:
+```
+#define SPDLOG_LEVEL_TRACE 0
+#define SPDLOG_LEVEL_DEBUG 1
+#define SPDLOG_LEVEL_INFO 2
+#define SPDLOG_LEVEL_WARN 3
+#define SPDLOG_LEVEL_ERROR 4
+#define SPDLOG_LEVEL_CRITICAL 5
+#define SPDLOG_LEVEL_OFF 6 
+```
+
+The type of tracing information logged for each of these levels is evident from their names. Comparing with the different [verbose modes](https://oneapi-src.github.io/oneDNN/dev_guide_verbose.html?highlight=onednn_verbose#run-time-controls) defined for oneDNN, these logging levels can be aligned with the verbose modes as follows:
+
+|   | `spdlog` Level        | oneDNN Verbose Mode                                              |
+|---|-----------------------|------------------------------------------------------------------|
+| 0 | SPDLOG_LEVEL_TRACE    | `all`                                                            |
+| 1 | SPDLOG_LEVEL_DEBUG    | `dispatch`, `debuginfo=<level>`, `check`                         |
+| 2 | SPDLOG_LEVEL_INFO     | `profile_create`, `profile_exec`, `profile`                      |
+| 3 | SPDLOG_LEVEL_WARNING  | ---                                                              |
+| 4 | SPDLOG_LEVEL_ERROR    | `error`                                                          |
+| 5 | SPDLOG_LEVEL_CRITICAL | -                                                                |
+| 6 | SPDLOG_LEVEL_OFF      | `none`                                                           |
+
+With this alignment, the tracing information printed out for each verbose mode can be also logged at the aligned level. Obviously, the logging level here is determined from the value of the `ONEDNN_VERBOSE` variable. 
+To exercise better control over the logged data, it is recommended to introduce another variable `ONEDNN_LOG_VERBOSE` which specifies the verbose mode for the logged data and also additionally allows the user to use the [`filter`]((https://oneapi-src.github.io/oneDNN/dev_guide_verbose.html?highlight=onednn_verbose#run-time-controls)) option for the verbose mode to filter the information being logged for supported components and primitives. 
```

Comment:
Using ONEDNN_LOG_VERBOSE is proposed so that the verbose levels can be separately specified for the logfiles and the console. 
For example, setting ONEDNN_VERBOSE=error and ONEDNN_LOG_VERBOSE=all will print all data to the logfiles but only display errors on the console.

Reply:
I'm not aware of solutions that would provide an information both to stdout and file, this might be misleading for an end-user to see stdout messages when logging into a file was requested.

Is there a specific request to support such mode?

Reply:
There is no specific ask for printing to both stdout and logfiles, but recommended solutions provide an option for the user to enable outputs to the console while logging.